builtin/providers/azurerm/resource_arm_managed_image.go

```diff
@@ -0,0 +1,421 @@
+package azurerm
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmManagedImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagedImageCreate,
+		Read:   resourceArmManagedImageRead,
+		Update: resourceArmManagedImageCreate,
+		Delete: resourceArmManagedImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_virtual_machine_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"os_disk.os_disk_ostype"},
+			},
+
+			"os_disk": {
```

Comment:
We will once the snapshots (separate ARM resource) are implemented. Snapshots are on the backlog at the moment, but should not be very hard to implement. Implementing snapshots could also potentially aid in automating our tests, since we could create snapshots from other subscriptions.